core/procedural-primitives/procedural-primitives.zig

```diff
@@ -185,12 +185,16 @@ pub fn createCubePrimitive(allocator: std.mem.Allocator, size: f32) !Primitive {
 }
 
 pub fn createCylinderPrimitive(allocator: std.mem.Allocator, radius: f32, height: f32, num_sides: u32) !Primitive {
-    var vertex_data = try std.ArrayList(VertexData).initCapacity(allocator, 500);
-    var index_data = try std.ArrayList(u32).initCapacity(allocator, 500);
+    const alloc_amt_vert: u32 = (num_sides * 12 + 12) / 3;
+    const alloc_amt_idx: u32 = num_sides * 12 + 12;
+
+    var vertex_data = try std.ArrayList(VertexData).initCapacity(allocator, alloc_amt_vert);
+    var index_data = try std.ArrayList(u32).initCapacity(allocator, alloc_amt_idx);
     var indexes: u32 = 0;
 
-    var temp_normal: [500 * 3]f32 = undefined;
-    var temp_vertices: [(500 * 3) + 3]f32 = undefined;
+    // (50 * 12 + 12)/3 = 204 (allocate for 50 side)
+    var temp_normal: [204 * 3]f32 = undefined;
```

Comment:
What happens if num_size is more than 50? And why do you need temp_normal/temp_vertices?

Reply:
There little bit of dilemma there. I used temp_vertices and temp_normal, because I think it would result in faster code as we are only processing a single array with all it values (necessary for manipulation) laid contagiously, so it would result in less cache misses. I haven't actually measured how much it improves tho lol. But if we don't care about performance (@slimsag choice) then I can just vertex_data/index_data directly.

The reason I fixed the sides to 50 is because I think it high enough, as in if you were to use smooth shading, you wouldn't notice any edges and it would look smooth. We can change that to whatever you want.

Reply:
> There little bit of dilemma there. I used temp_vertices and temp_normal, because I think it would result in faster code as we are only processing a single array with all it values (necessary for manipulation) laid contagiously, so it would result in less cache misses. I haven't actually measured how much it improves tho lol. But if we don't care about performance (@slimsag choice) then I can just vertex_data/index_data directly.
> 
> The reason I fixed the sides to 50 is because I think it high enough, as in if you were to use smooth shading, you wouldn't notice any edges and it would look smooth. We can change that to whatever you want.

Using MultiArrayList solves these problems